src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V62/Eth62ProtocolHandler.cs

```diff
@@ -236,7 +239,48 @@ private void Handle(StatusMessage status)
 
         protected void Handle(TransactionsMessage msg)
         {
-            IList<Transaction> transactions = msg.Transactions;
+            IList<Transaction> iList = msg.Transactions;
+
+            if (iList is ArrayPoolList<Transaction> apl)
+            {
+                HandleFast(apl.AsSpan());
+            }
+            else if (iList is Transaction[] array)
+            {
+                HandleFast(array.AsSpan());
+            }
+            else if (iList is List<Transaction> list)
+            {
+                HandleFast(CollectionsMarshal.AsSpan(list));
+            }
+            else
+            {
+                HandleSlow(iList);
+            }
+        }
+
+        private void HandleFast(Span<Transaction> transactions)
+        {
+            bool isTrace = Logger.IsTrace;
+            for (int i = 0; i < transactions.Length; i++)
+            {
+                Transaction tx = transactions[i];
+                tx.DeliveredBy = Node.Id;
+                tx.Timestamp = _timestamper.UnixTime.Seconds;
+                AcceptTxResult accepted = _txPool.SubmitTx(tx, TxHandlingOptions.None);
+                _floodController.Report(accepted);
+
+                if (isTrace) Log(tx, accepted);
+            }
+
+            void Log(Transaction tx, AcceptTxResult accepted)
+            {
+                Logger.Trace($"{Node:c} sent {tx.Hash} tx and it was {accepted} (chain ID = {tx.Signature?.ChainId})");
+            }
```

Comment:
Code duplication between HandleFast and Handle slow

Reply:
Resolved